src/GafferUI/ScaleHandle.cpp

```diff
@@ -113,8 +113,11 @@ Imath::V3f ScaleHandle::scaling( const DragDropEvent &event )
 		// When performing uniform scales, the handle is at the origin, so the
 		// pattern we use above gets very twitchy. We instead need to treat the
 		// click point as scale=1 and relative movement in +ve x as a scale
-		// increase and anything in -ve x as a scale decrease.
-		scale = m_drag.updatedPosition( event ) - m_drag.startPosition();
+		// increase and anything in -ve x as a scale decrease. Coordinates are in
+		// gadget-space, which does not scale by camera position. Normalize
+		// by `rasterScaleFactor()` to prevent very large scaling when zoomed out
+		// and small scaling when zoomed in.
+		scale = m_drag.updatedPosition( event ) / rasterScaleFactor().x;
```

Comment:
Good work tracking this down!

I do think we need to keep doing this relative to the drag start position though (but with your raster fix still of course). The handle is big enough (especially after using the `+` button to enlarge it) that you do get a little jump at the start of the drag if the start position isn't taken into account.

I also think the fact that we're taking just the `.x` of the raster scale is a warning sign that this solution isn't quite as general as it should be. It works for the ScaleTool because in that case the ScaleHandle's own scaling (in its gadget transform) is uniform, but that is not the general case. If the handle's scaling was non-uniform, then we'd be in trouble. If you look at the implementation of `rasterScaleFactor()`, you'll see that it corrects for two things - the distance from camera, and scaling in the handle's transform. In this case we only want to correct for the first.

It seemed weird that we were doing a gadget-space drag at all, given that we really want the scale to be proportional to raster-space movement. So I looked a little into the history of this. In fact, we did originally implement this purely using raster space, but then in https://github.com/GafferHQ/gaffer/pull/3438/commits/97f62bffc048f73ff1ab004afedb47c3e755d030 we reimplemented in this slightly weird way, so that we could take advantage of the "precise mode" (shift + drag) provided by `Handle::LinearDrag`.

I'd propose that the more general fix would be to keep using LinearDrag as we are, but to use `gadgetToRasterSpace()` to recover the raster space positions we want from `drag.startPosition()` and `drag.updatedPosition()` and then to use the original logic from before that commit (so we also take into account the viewport size).

While we're here, do we think that there's any benefit in allowing negative scaling in this case? To me it doesn't seem that useful, but it does make it harder to scale something down close to zero without accidents...

Reply:
> I also think the fact that we're taking just the `.x` of the raster scale is a warning sign that this solution isn't quite as general as it should be

Ah right, the gadget scale compensation can make make it non-uniform.

> I'd propose that the more general fix would be to keep using LinearDrag as we are, but to use `gadgetToRasterSpace()` to recover the raster space positions we want from `drag.startPosition()` and `drag.updatedPosition()` and then to use the original logic from before that commit (so we also take into account the viewport size).

Cool, that makes sense.

> do we think that there's any benefit in allowing negative scaling in this case

For uniform scale, maybe not? If we prevent that, users could always manually negate all the scales if they want.

> it does make it harder to scale something down close to zero without accidents...

Maybe we map positive raster differences as linear scale factors, and negative values approach zero but never get there? The first function in this [GraphToy](https://graphtoy.com/?f1(x,t)=1/(1-x)&v1=true&f2(x,t)=x+1&v2=true&f3(x,t)=&v3=true&f4(x,t)=&v4=true&f5(x,t)=&v5=false&f6(x,t)=&v6=true&grid=1&coords=0,0,7.451055876709868) might work, and is continuous with a linear function at 0. I'd have to try it to see how it works in practice.

Reply:
> The first function in this GraphToy might work

Sounds worth a bash

Reply:
> I do think we need to keep doing this relative to the drag start position though

Yes, when the handle is bigger that subtraction becomes important. Fixed in 1c6b08c9e9c64d055051dc25fdaf9710e13173f8

Reply:
> Sounds worth a bash

I gave it a try and it seems pretty buttery! I like it especially for lights - using the uniform scale on a quad light could cause it to flip pretty easily, this gives more control when making it smaller.